src/Transports/MassTransit.ActiveMqTransport/Configuration/Configurators/ActiveMqBusFactoryConfigurator.cs

```diff
@@ -97,5 +99,44 @@ public override IEnumerable<ValidationResult> Validate()
             if (string.IsNullOrWhiteSpace(_settings.EntityName))
                 yield return this.Failure("Bus", "The bus queue name must not be null or empty");
         }
+
+        /// <summary>
+        /// New configurable extensionpoint, available/accessble via the configurator.
+        /// This extension point can be used to enable artimis compatibility
+        /// It allows to override the generated BindingConsumeTopologySpecification
+        /// That specification is responsible (on calling its Apply method) for interacting with an instance of IReceiveEndpointBrokerTopologyBuilder.
+        /// This allows the specification to control what queues, bindings, are created.
+        /// For the interop with Artemis the name of the consumer queue is important
+        /// The original specification was : ActiveMqBindConsumeTopologySpecification
+        /// a new one has been already provided for Artemis => ArtemisBindConsumeTopologySpecification
+        ///
+        /// When you call EnableArtemisCompatibility() => this factory method is automatically initialized with a factory method that will
+        /// create a ArtemisBindConsumeTopologySpecification
+        ///
+        /// This extension could also be used to create your own specifications if you like other behavior for creating queues and bindings or
+        /// name conventions
+        /// </summary>
+        public ActiveMqBindingConsumeTopologySpecificationFactoryMethod BindingConsumeTopologySpecificationFactoryMethod
+        {
+            get => _busConfiguration.BindingConsumeTopologySpecificationFactoryMethod;
+            set => _busConfiguration.BindingConsumeTopologySpecificationFactoryMethod = value;
+        }
+
+        /// <summary>
+        /// Handy shortcut utility function for enabling Artemis compatibility.
+        /// </summary>
+        public void EnableArtemisCompatibility()
+        {
+            BindingConsumeTopologySpecificationFactoryMethod = (string topic) =>
+            {
+                return new ArtemisBindConsumeTopologySpecification(topic);
+            };
+        }
+
+        public void UpdateReceiveQueueName(Func<string, string> transformReceiveQueueName)

```

Comment:
this is not pretty, or easily understood, nor globally applicable.

Reply:
new commits have different implementation